qupath-core/src/main/java/qupath/lib/common/GeneralTools.java

```diff
@@ -775,7 +782,18 @@ public static boolean deleteFile(File fileToDelete, boolean preferTrash) {
 		return fileToDelete.delete();
 	}
 
-	private static boolean moveToTrash(Desktop desktop, File fileToDelete) {
+	/**
+	 * Attempt to move the provided file or directory to trash. This may fail depending on the current platform.
+	 *
+	 * @param desktop the desktop to use when moving the provided file/directory to trash
+	 * @param fileToDelete the file or directory to move to trash
+	 * @return whether the provided file/directory was moved to trash
+	 * @throws NullPointerException if one of the provided parameters is null
+	 * @throws SecurityException if the calling thread doesn't have the permission to read or move the provided file/directory
+	 * @throws UnsupportedOperationException if the provided desktop doesn't support the {@link Desktop.Action#MOVE_TO_TRASH} action
```

Comment:
Shouldn't we just return `false` here?

I think a public version of this method shouldn't expose that it is using `Desktop` at all. For example, in the future we might switch to [JNA Platform](https://javadoc.io/doc/net.java.dev.jna/jna-platform/latest/com/sun/jna/platform/FileUtils.html#moveToTrash(java.io.File...)).

We could therefore have
```java
private static boolean moveToTrash(Desktop desktop, File fileToDelete) // as before

public static boolean moveToTrash(File fileOrDirectory) // new public method
```